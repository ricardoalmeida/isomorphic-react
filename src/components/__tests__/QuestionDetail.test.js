import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail';
import React from "react";
import renderer from "react-test-renderer";

describe('The test detail component', () => {
    describe('The container element', () => {
        describe("mapStateToProps", () => {
            it('should map state to props correctly', () => {
                const sampleQuestion = {
                    question_id: 42,
                    body: "Space is big",
                };
                const appState = {
                    questions: [sampleQuestion],
                };
                const ownProps = {
                    question_id: 42
                };
                const componentState = mapStateToProps(appState, ownProps);
                expect(componentState).toEqual(sampleQuestion);
            });
        })
    })
    describe('The display element', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <QuestionDetailDisplay
                    title="The meaning of life"
                    body="42"
                    answer_count={0}
                    tags={[`hit`]}
                />
            );
            expect(tree.toJSON).toMatchSnapshot();
        });
    });
});