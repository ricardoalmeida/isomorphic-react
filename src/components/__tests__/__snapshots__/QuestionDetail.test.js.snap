// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The test detail component The display element should not regress 1`] = `[Function]`;
